joss_paper/paper.md

```diff
@@ -0,0 +1,208 @@
+Functions are simple in usage and pure in scope, and can be chained together to build customized analyses of data while estimators are implemented as classes that follow the tried-and-tested scikit-learn API [@sklearn2013api].
+General and robust workflows following best practices are also provided, which can be used as reference implementations and examples.
+
+First and foremost, scientific code must be correct and we try to ensure this requirement by following best software engineering practices during development, close to full test coverage of all code in the library (currently 99%), and providing citations to published papers for included algorithms. 
+We use a curated, public data set ([*alchemtest*](https://github.com/alchemistry/alchemtest)) for automated testing; code in *alchemtest* is published under the open source BSD-3 clause license while all data are included under an [open license](https://opendefinition.org/licenses/#recommended-conformant-licenses) such as [CC0](https://creativecommons.org/publicdomain/zero/1.0/) (public domain) or [CC-BY](http://opendefinition.org/licenses/cc-by/) (attribution required).
+
+The guiding design principles are summarized as:
+
+1. Use functions when possible, classes only when necessary (or for estimators, see (2)).
+2. For estimators, mimic the object-oriented scikit-learn API as much as possible.
+3. Aim for a consistent interface throughout, e.g. all parsers take similar inputs and yield a common set of outputs, using the `pandas.DataFrame` as the underlying data structure.
+4. Have *all* functionality tested.
+
+*alchemlyb* supports recent versions of Python 3 and follows the [SPEC 0 (Minimum Supported Dependencies)](https://scientific-python.org/specs/spec-0000/) Scientific Python Ecosystem Coordination community standard for deciding on when to drop support for older versions of Python and dependencies.
+Releases are numbered following the [Semantic Versioning 2.0.0](https://semver.org/) standard of MAJOR.MINOR.PATCHLEVEL, which ensures that users immediately understand if a release may break backwards compatibility (increase of the major version), adds new features (increase of minor version), or only contains bug fixes or other changes that do not directly affect users.
+All code is published under the open source BSD-3 clause license.
+
+## Library structure
+
+*alchemlyb* offers specific parsers in `alchemlyb.parsing` to load raw free energy data from various molecular simulation packages ([GROMACS](https://www.gromacs.org/) [@Abraham2015aa], [Amber](https://ambermd.org/) [@Case2005uq], [NAMD](https://www.ks.uiuc.edu/Research/namd/) [@phillips2020scalable], and [GOMC](https://gomc-wsu.org/) [@Nejahi2021aa]) and provides a general structure for implementing parsers for other packages that are not yet supported.
```

Comment:
I decided to leave the citations in here so that one can easily read the sections independently; personally I also like being able to just go references quickly.